### PR TITLE
=pro #17570 upgrade scalariform plugin

### PR DESCRIPTION
**Pull request by jrudolph**
I tried adapting the settings a bit to keep the old style as much as possible with these changes:

``` diff
-      .setPreference(AlignParameters, true)
+      .setPreference(AlignParameters, false)
+      .setPreference(SpacesAroundMultiImports, true)
+      .setPreference(DanglingCloseParenthesis, Prevent)
```

which mostly keeps the existing formatting (aside from those in these 300 files...). Some changes are only due to the fact the old scalariform was broken and didn't format lots of files for quite some time it seems.

On the other hand, it may make sense to try to use more of the default settings right now and do the hard and breaking change right now). Especially, `AlignParameters` actually looks quite nice for case classes if enabled (although it also generally increases conflict risk similar to alignment of case-lines in patterns).

### CODE REVIEW

**Comment by jrudolph** [comment]
It seems there's still a bug in scalariform wrt parsing `val ->` in FSM.scala. See https://github.com/daniel-trinh/scalariform/issues/113.

**Comment by jrudolph** [comment]
There's https://github.com/scala-ide/scalariform/issues/202 which, when fixed, would prevent a lot of those changes needed here.

The test failure is also a genuine problem. Pattern matches using FSM's `->` are rewritten to use `→` which isn't defined.

**Comment by ktoso** [comment]
Status update: We'll focus on merging pending stuff and do the plugin upgrade soon

**Comment by jrudolph** [comment]
It may make sense to wait for a while. It may be that the next version of sbt-scalariform fill fix most of the otherwise required changes.

**Comment by ktoso** [comment]
We do want to do this soon however for now I'll close the PR (need to keep their number a bit more managable, 30+ is hard to look at :-))

File: project/plugins.sbt [hunk 1]
```diff
@@ -11,7 +11,8 @@ resolvers += "Bintray Jcenter" at "https://jcenter.bintray.com/"
 addSbtPlugin("com.typesafe.sbt" % "sbt-multi-jvm" % "0.3.8")
 //#sbt-multi-jvm
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-scalariform" % "1.2.0")
+addSbtPlugin("org.scalariform" % "sbt-scalariform" % "1.6.0")
+libraryDependencies += "org.scalariform" %% "scalariform" % "0.2.0-SNAPSHOT"
```
Comment (ktoso) [inline project/plugins.sbt]: not sure if we want to be on snapshot :~

**Pull request closed without merging**